### ISSUE TITLE
Edit: Split preview idea

### ISSUE THREAD

**Issue by jenstornell** [distantnative/panel-bar#31]
Here is just an idea, probably hard to get working. A split screen preview. Think about a "Preview" menu item in the panelbar.

### How it could work

Click on the "Preview" menu in the panelbar.

An iframe is placed to the left over 50% of the screen. Inside it is the editable page loaded by the panel.
Another iframe is placed to the right over 50% of the screen. Inside it is the frontend.

When clicking save or changing a field, the right (frontend) iframe is updated.

### Why?

It's nice to to something that feels live. Update a page on the fly. It also has no depencences. I mean it does not need any special fancy placeholders or advanced JS to keep them in sync.

In short, it just needs to know when something is changed in the left iframe and then update the right one.

### Possible?

I don't know it's possible to access change events in an iframe like that. If not this will not work.

### For the panelbar?

If you think it's too big for the panelbar or simple don't fit, I might try to do a panelbar plugin out of it.

Feedback?

**Comment by distantnative** [comment]
I'll play around with this idea a bit and decide then whether it's a fit or not

**Comment by distantnative** [comment]
I'm facing the problem, that the current panel isn't  very iframe-friendly as it is not possible to scroll it when displayed in an iframe

**Comment by jenstornell** [comment]
I got it scrollable kirby panel. Have you set the scrolling attribute?
http://www.w3schools.com/tags/tryit.asp?filename=tryhtml_iframe_scrolling

I also added this to the iframe:

    position: fixed;
    width: 50%;
    height: 100%;
    z-index: 300;

Seems to work.

**Comment by distantnative** [comment]
It's a quite hackish thing to do, but you can find a first (seemingly working) version on the respective feature branch

**Comment by jenstornell** [comment]
Sorry about the delay. I work tuesdays, thursdays (half day) and fridays.

I updated the files but could not get it to work. I tried both download by the download button and each file as raw. The preview button does not appear.

**Comment by distantnative** [comment]
Yes, you need enhancedJS (=> jQuery) for preview, otherwise it doesn't make sense to even show the button atm. 

Might be able to rewrite it less-jQuery to a later point in time.

And the top undefined issue should be fixed.

**Comment by jenstornell** [comment]
Should I add all problems as issues? I have some...

1. I had to add position: fixed; to .panelbar__preview--view to have it shown.
1. When shrinking the Window it disapears again. I don't know if a div wrapper is needed around each iframe.
1. On save the right window is not updated.
1. The panelbar is also on the right preview. Would it be nice to with JS / CSS hide it? in the iframe?
1. The sidebar toggle (when small window) goes below it's own bar, probably because of the iframe.
1. I don't like the delay-thing in the JS but that might be needed for it to work?

Except for all this, i still love the idea and think it could work well without adding special code to the admin or template.

**Comment by distantnative** [comment]
>Here is more of a question. Is the "Preview" feature to big to be built into the panel-bar project? I wrote down some issues in a previous comment.
>
>I have found even more problems like:
>
>When panel-bar at the bottom the CSS might not work as expected.
When page is updated there is a flickering (blinking). I don't mind a delay of the updated content but the blinking makes me blink. I think it's possible to go around with some more work, like only update the body with updated HTML. That would require some Ajax as well.
So, will the preview feature be to big for the panel-bar project? Should it be made as a standalone plugin? Or should it be made as a standalone files in this plugin to separate them, to avoid messyness?
>
>If you feel it's too big I can try to build it as an own plugin but if you still want to continue the development I will leave it to you. Just let me know.

**Comment by distantnative** [comment]
I decided that I'm quite happy with the current fullscreen iframe edit mode and don't want to get into the really hackish side-by-side split view thing. at least not soon.

**Comment by jenstornell** [comment]
I agree with you at this point. It's hackish, might not work well and Bastian might even do a built in preview in the panel, maybe. If he does I will vote for a default panel-bar preview link element. :)
